dev-tools/check_duplicate_test_data.php

```diff
@@ -0,0 +1,152 @@
+            continue;
+        }
+
+        $exceptionsBecauseOfMoreThanOneDuplicate = [ // should only shrink
+            'PhpCsFixer\Tests\Console\Command\SelfUpdateCommandTest::provideExecuteCases',
+            'PhpCsFixer\Tests\Console\Output\Progress\DotsOutputTest::provideDotsProgressOutputCases',
+            'PhpCsFixer\Tests\Fixer\ArrayNotation\TrimArraySpacesFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\Basic\BracesFixerTest::provideFixClassyBracesCases',
+            'PhpCsFixer\Tests\Fixer\Basic\BracesPositionFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\Basic\CurlyBracesPositionFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\ClassNotation\FinalClassFixerTest::provideFix80Cases',
+            'PhpCsFixer\Tests\Fixer\Basic\PsrAutoloadingFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\CastNotation\LowercaseCastFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\ClassNotation\FinalClassFixerTest::provideFix82Cases',
+            'PhpCsFixer\Tests\Fixer\ControlStructure\NoUnneededControlParenthesesFixerTest::provideFixAllCases',
+            'PhpCsFixer\Tests\Fixer\ControlStructure\NoUselessElseFixerTest::provideFixIfElseIfElseCases',
+            'PhpCsFixer\Tests\Fixer\ControlStructure\NoUselessElseFixerTest::provideFixIfElseCases',
+            'PhpCsFixer\Tests\Fixer\ControlStructure\YodaStyleFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\ControlStructure\YodaStyleFixerTest::providePHP71Cases',
+            'PhpCsFixer\Tests\Fixer\DoctrineAnnotation\DoctrineAnnotationArrayAssignmentFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\DoctrineAnnotation\DoctrineAnnotationArrayAssignmentFixerTest::provideFixWithColonCases',
+            'PhpCsFixer\Tests\Fixer\DoctrineAnnotation\DoctrineAnnotationSpacesFixerTest::provideFixAroundParenthesesOnlyCases',
+            'PhpCsFixer\Tests\Fixer\LanguageConstruct\SingleSpaceAfterConstructFixerTest::provideFixWithUseCases',
+            'PhpCsFixer\Tests\Fixer\LanguageConstruct\SingleSpaceAroundConstructFixerTest::provideFixWithUseCases',
+            'PhpCsFixer\Tests\Fixer\PhpUnit\PhpUnitStrictFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Fixer\Phpdoc\PhpdocInlineTagNormalizerFixerTest::provideFixCases',
+            'PhpCsFixer\Tests\Tokenizer\Analyzer\AlternativeSyntaxAnalyzerTest::provideItThrowsOnInvalidAlternativeSyntaxBlockStartIndexCases',
+            'PhpCsFixer\Tests\Tokenizer\Analyzer\FunctionsAnalyzerTest::provideIsGlobalFunctionCallCases',
+            'PhpCsFixer\Tests\Tokenizer\TokenTest::provideIsMagicConstantCases',
+            'PhpCsFixer\Tests\Tokenizer\TokensAnalyzerTest::provideIsBinaryOperatorCases',
+        ];
+        if (in_array($testClassName.'::'.$method->getName(), $exceptionsBecauseOfMoreThanOneDuplicate, true)) {
+            continue;
+        }
+
+        $exceptionsBecauseOfOneDuplicate = [ // should only shrink
```

Comment:
Does it actually make sense to distinguish exceptions by one or more duplicates? If class is refactored (not using this script, but because someone finds something out while working on something) and duplicates are removed, it may happen that exception defined in `$exceptionsBecauseOfMoreThanOneDuplicate` should land into `$exceptionsBecauseOfOneDuplicate` - isn't it unneeded work and room for getting out of sync? Maybe one list is better?

Reply:
Data providers with one duplicate are much easier to fix, this is just for convenience. If I do not manage to fix some provider I will create a general `$exceptions` list.